app/controllers/decidim/challenges/challenges_controller.rb

```diff
@@ -21,8 +21,6 @@ def index; end
 
       def show
         @challenge = Challenge.find(params[:id])
```

Comment:
Maybe here, you could initialize this deleted vars with some conditionals.
E.g.:
```
@sdg = @challenge.sdg_code if @challenge.sdg_code.present?
@sdg_index = (1 + Decidim::Sdgs::Sdg.index_from_code(@challenge.sdg_code.to_sym)).to_s.rjust(2, "0") if @sdg
```
or something like that...